### PR TITLE
Search by external_key

### PR DESCRIPTION
**Pull request by luismorenolopera**
- In the django admin, EDUNEXT OPENEDX MULTITENANCY section, Routes model now allows searching by external_key.
- Fix bug in class TenantConfigAdmin.
-  Template dir: now show hierarchy, name <- parent <- grandparent.

### CODE REVIEW

File: eox_tenant/admin.py [hunk 1]
```diff
@@ -106,7 +104,13 @@ def template_dir(self, tenant_config):
         """
         # pylint: disable=broad-except
         try:
-            return tenant_config.lms_configs.get("template_dir", "NOT CONFIGURED")
+            themes = ["name", "parent", "grandparent"]
+            theme_config = tenant_config.theming_configs.get("THEME_OPTIONS", "theme")
```
Comment (andrey-canon) [inline eox_tenant/admin.py]:
```suggestion
            theme_config = tenant_config.theming_configs.get("THEME_OPTIONS", {})
```

File: eox_tenant/admin.py [hunk 2]
```diff
@@ -106,7 +104,13 @@ def template_dir(self, tenant_config):
...
+            result = ""
+            for key in themes:
+                if key in theme_config["theme"]:
```
Comment (andrey-canon) [inline eox_tenant/admin.py]:
```suggestion
                if key in theme_config.get("theme", {}):
```

File: eox_tenant/admin.py [hunk 3]
```diff
@@ -106,7 +104,13 @@ def template_dir(self, tenant_config):
         """
         # pylint: disable=broad-except
         try:
-            return tenant_config.lms_configs.get("template_dir", "NOT CONFIGURED")
+            themes = ["name", "parent", "grandparent"]
```
Comment (andrey-canon) [inline eox_tenant/admin.py]: rename, maybe theme_keys

File: eox_tenant/admin.py [hunk 4]
```diff
@@ -106,7 +104,13 @@ def template_dir(self, tenant_config):
         """
         # pylint: disable=broad-except
         try:
-            return tenant_config.lms_configs.get("template_dir", "NOT CONFIGURED")
+            themes_keys = ["name", "parent", "grandparent"]
+            theme_config = tenant_config.theming_configs.get("THEME_OPTIONS", {})
+            result = ""
+            for key in themes_keys:
```
Comment (andrey-canon) [inline eox_tenant/admin.py]:
```suggestion
           theme = theme_config.get("theme", {})
           values = [theme[key] for key in themes_keys if key in theme]

           if values:
               return " <- ".join(values)
           else 
               return ""
```

File: eox_tenant/admin.py [hunk 5]
```diff
@@ -106,7 +104,13 @@ def template_dir(self, tenant_config):
...
         # pylint: disable=broad-except
         try:
-            return tenant_config.lms_configs.get("template_dir", "NOT CONFIGURED")
+            themes_keys = ["name", "parent", "grandparent"]
+            theme_config = tenant_config.theming_configs.get("THEME_OPTIONS", {})
+            theme = theme_config.get("theme", {})
+            values = [theme[key] for key in themes_keys if key in theme]
+            if values:
+                return " <- ".join(values)
+            return ""
```
Comment (andrey-canon) [inline eox_tenant/admin.py]: Set this as "NOT CONFIGURED"

File: eox_tenant/admin.py [hunk 6]
```diff
@@ -41,11 +39,18 @@ def sitename(self, microsite):
...
     def template_dir(self, microsite):
         """
-        TODO: add me
+        Read only method to calculate template dir attribute from config model.
         """
         # pylint: disable=broad-except
         try:
-            return microsite.values.get('template_dir', "NOT CONFIGURED")
+            themes_keys = ["name", "parent", "grandparent"]
+            theme_config = TenantConfig.objects.get(external_key=microsite.key)
```
Comment (andrey-canon) [inline eox_tenant/admin.py]: you should use microsites.values not tenantconfig

**Comment by andrey-canon** [comment]
@luismorenolopera  you repeat the code for microsites and tenantconfig, you could do that in a general method.

**Comment by felipemontoya** [comment]
This PR has 3 different change sets. I suggest we close it and open a different one for each necessary point

**Pull request closed without merging**